server/stream.go

```diff
@@ -5513,15 +5516,30 @@ func (mset *stream) hasAllPreAcks(seq uint64, subj string) bool {
 // Check if we have all consumers pre-acked.
 // Write lock should be held.
 func (mset *stream) clearAllPreAcks(seq uint64) {
-	delete(mset.preAcks, seq)
+	for o, ss := range mset.preAcks {
+		if ss.Delete(seq) && ss.Size() == 0 {
+			delete(mset.preAcks, o)
+		}
+	}
 }
 
 // Clear all preAcks below floor.
 // Write lock should be held.
 func (mset *stream) clearAllPreAcksBelowFloor(floor uint64) {
-	for seq := range mset.preAcks {
-		if seq < floor {
-			delete(mset.preAcks, seq)
+	for o, ss := range mset.preAcks {
+		min, max := ss.MinMax()
```

Comment:
This is dynamically calculated on the fly and not maintained.

Reply:
Good point, I wonder if this outweighs the benefit of the following two checks...